docs/Compile.rst

```diff
@@ -121,7 +156,9 @@ DF, which causes DF to use your system libstdc++ instead::
 Note that distributing binaries compiled with newer GCC versions requires end-
 users to delete libstdc++ themselves and have a libstdc++ on their system from
 the same GCC version or newer. For this reason, distributing anything compiled
-with GCC versions newer than 4.8 is discouraged.
```

Comment:
OK I'm a bit confused about what needs to be said here.

My understanding was that:
* If we're distributing, then currently only 4.5 should be used until such time as we look at libstdc++ distribution
* If it's for personal, anything >= 4.5 is fine, right up to current 5.2.1 and, according to GCC's stated ABI compat, beyond for the foreseeable future
  *  and there's a strong benefit to using a later version on modern distros, simply in that it's installed by default and one doesn't need to spend the time working out how to install an older version, mess with the alternatives system, etc.

I'm not sure where a 4.7 or 4.8 or anything-other-than-4.5 fits in?  If you have to install an old version of GCC, surely it's only for distribution purposes and therefore must be 4.5?  And if you're not distributing, you can use any later version and it's easier for you if you do so.  

So if a user happens to be on a 4.7, 4.8 or 4.9 system then that's fine, just as it is if they're on 5.x, but there's no reason anyone would want to install 4.x unless it's 4.5 for distributing?  So is there a need to mention any version besides 4.5 still?

(As an aside, it just occurred to me that it might be useful to have a new "Note on using GCC 5.x" section which explicitly mentions the ABI compatibility flag that cmake is applying.  This relates to potential new contributors, who must be aware that they need to compile their own plugins with that flag if they're testing on GCC 5 and plan to work with DFHack.  That might belong in Contributing.rst, instead of or as well as this Compile.rst,)